taskipy/cli.py

```diff
@@ -12,14 +12,18 @@ def main():
         prog='task',
         description='runs a task specified in your pyproject.toml under [tool.taskipy.tasks]',
     )
-    parser.add_argument('name', help='name of the task')
-    parser.add_argument(
-        'args', nargs=argparse.REMAINDER, help='arguments to pass to the task'
-    )
+    parser.add_argument('--list', help='show list of available tasks', action='store_true')
+    parser.add_argument('name', help='name of the task', nargs='?')
+    parser.add_argument('args', nargs=argparse.REMAINDER, help='arguments to pass to the task')
     args = parser.parse_args()
-
     try:
-        exit_code = TaskRunner(Path.cwd() / 'pyproject.toml').run(args.name, args.args)
+        runner = TaskRunner(Path.cwd() / 'pyproject.toml')
+        if args.list:
+            longest = len(max(runner.project.tasks, key=len))
+            for k, v in runner.project.tasks.items():
```

Comment:
just a proposal here, but perhaps we should consider moving this logic into `TaskRunner`? :) That way, we can keep all the task handling logic in one place.